protocol/app/process/transactions.go

```diff
@@ -4,20 +4,37 @@ import (
 	errorsmod "cosmossdk.io/errors"
 	abci "github.com/cometbft/cometbft/abci/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	"slices"
 )
 
-type txtype int
-
 const (
-	ProposedOperationsTxType txtype = 1
-	AcknowledgeBridgesTxType txtype = 2
-	AddPremiumVotesTxType    txtype = 3
-	UpdateMarketPricesTxType txtype = 4
+	minTxsCount                   = 4
+	proposedOperationsTxIndex     = 0
+	updateMarketPricesTxLenOffset = -1
+	addPremiumVotesTxLenOffset    = -2
+	acknowledgeBridgesTxLenOffset = -3
+	lastOtherTxLenOffset          = acknowledgeBridgesTxLenOffset
+	firstOtherTxOffset            = 1
```

Comment:
for naming consistency, this should be `TxIndex` instead of `TxOffset`?

Reply:
I used `TxIndex` for cases where it is a raw index while `TxLenOffset` to indicate its an offset of the `len` of the `Tx` list. I can swap to just use `TxOffset`. WDYT?

Reply:
I agree with the naming convention, I am talking in particular about `firstOtherTxOffset` which I think is really an Index and not an Offset of some sort.

For example, imagine adding another mandatory transaction that must come second. Then we would change the `firstOtherTxOffset` variable to be `2`.

Reply:
SGTM